src/BlockTypes/LegacyTemplate.php

```diff
@@ -180,4 +184,22 @@ protected function render_archive_product() {
 		wp_reset_postdata();
 		return ob_get_clean();
 	}
+
+	/**
+	 * Get HTML markup with the right classes by attributes.
+	 *
+	 * @param string $content Block content.
+	 * @param array  $block Parsed block data.
+	 * @return string Rendered block type output.
+	 */
+	public function get_markup_with_classes_by_attributes( string $content, array $block ) {
+		$pattern     = '/(?<=class=\")[^"]+(?=\")/';
```

Comment:
> The global/wrapper-start.php can be overridden in the theme, so the wrapper may not have the class attribute. 
Great catch! I don't think about this case.

>but a note for the theme developer will be nice here.

Do you mean to add a comment or what else?

Reply:
> We don't need to cover that case in our logic

I want to clarify on this. I meant that we may not need to address that issue in our logic if it takes too much effort because it's a rare case. Personally, I never see any wrapper that doesn't have class. But if you can solve it, go ahead ; ).

> Do you mean to add a comment or what else?

I thinking about adding a comment to your callback `get_markup_with_classes_by_attributes`.